bundles/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/factory/HueThingHandlerFactory.java

```diff
@@ -38,57 +41,74 @@
+                    HueBridgeHandler.SUPPORTED_THING_TYPES.stream(), HueLightHandler.SUPPORTED_THING_TYPES.stream(),
                     DimmerSwitchHandler.SUPPORTED_THING_TYPES.stream(), TapSwitchHandler.SUPPORTED_THING_TYPES.stream(),
                     PresenceHandler.SUPPORTED_THING_TYPES.stream(),
                     GeofencePresenceHandler.SUPPORTED_THING_TYPES.stream(),
                     TemperatureHandler.SUPPORTED_THING_TYPES.stream(), LightLevelHandler.SUPPORTED_THING_TYPES.stream(),
                     ClipHandler.SUPPORTED_THING_TYPES.stream(), HueGroupHandler.SUPPORTED_THING_TYPES.stream())
             .flatMap(i -> i).collect(Collectors.toUnmodifiableSet());
 
-    private final HttpClient httpClient;
+    private final HttpClientFactory httpClientFactory;
     private final HueStateDescriptionProvider stateDescriptionProvider;
+    private final Clip2StateDescriptionProvider clip2StateDescriptionProvider;
     private final TranslationProvider i18nProvider;
     private final LocaleProvider localeProvider;
+    private final ThingRegistry thingRegistry;
+    private final ItemChannelLinkRegistry itemChannelLinkRegistry;
 
     @Activate
     public HueThingHandlerFactory(final @Reference HttpClientFactory httpClientFactory,
             final @Reference HueStateDescriptionProvider stateDescriptionProvider,
-            final @Reference TranslationProvider i18nProvider, final @Reference LocaleProvider localeProvider) {
-        this.httpClient = httpClientFactory.getCommonHttpClient();
+            final @Reference Clip2StateDescriptionProvider clip2StateDescriptionProvider,
+            final @Reference TranslationProvider i18nProvider, final @Reference LocaleProvider localeProvider,
+            final @Reference ThingRegistry thingRegistry,
+            final @Reference ItemChannelLinkRegistry itemChannelLinkRegistry) {
+        this.httpClientFactory = httpClientFactory;
         this.stateDescriptionProvider = stateDescriptionProvider;
+        this.clip2StateDescriptionProvider = clip2StateDescriptionProvider;
         this.i18nProvider = i18nProvider;
         this.localeProvider = localeProvider;
+        this.thingRegistry = thingRegistry;
+        this.itemChannelLinkRegistry = itemChannelLinkRegistry;
     }
 
     @Override
     public @Nullable Thing createThing(ThingTypeUID thingTypeUID, Configuration configuration,
             @Nullable ThingUID thingUID, @Nullable ThingUID bridgeUID) {
-        if (HueBridgeHandler.SUPPORTED_THING_TYPES.contains(thingTypeUID)) {
+        if (HueBindingConstants.THING_TYPE_BRIDGE_API2.equals(thingTypeUID)) {
+            return super.createThing(thingTypeUID, configuration, thingUID, null);
+        } else if (Clip2ThingHandler.SUPPORTED_THING_TYPES.contains(thingTypeUID)) {
+            return super.createThing(thingTypeUID, configuration, thingUID, bridgeUID);
```

Comment:
You are not handling the case `thingTypeUID` is null.
I am not sure in which case it happens.

Reply:
> You are not handling the case thingTypeUID is null.

To be be fair to myself, you should really say _'whoever wrote the original code was not handling the case thingTypeUID is null'_ -- as I only added the lines 107..111

But putting that aside `thingTypeUID` is a method argument of a @NonNullByDefault class, so surely the null checks would be done elsewhere.

But perhaps you are referring to the code at line 175 ?? .. where the null case IS handled at line 203 -- or??

Reply:
In fact, it was handled at lines 114, 123 and 126 through the methods getLightUID, getSensorUID and getGroupUID, which all finally call the private method `getThingUID` when `thingUID` parameter was null.
The problem is not the parameter `thingTypeUID` (which is never null) from method `createThing` but the parameter `thingUID` which can be null. I just don't know in which circumstances it is null.
If the parameter is null, your code will simply call `super.createThing` with the null parameter (while an UID was determined in that case for old things). **We just have to check what it does in that case in the parent class (core framework).**
I am sure this case is supported and your code will work but it will probably work differently from the previous one.

Reply:
Umm. I have tested the code on live systems, (both OH 4.0 and OH 3.4.x) and never seen errors. => So is it OK to leave my code as it is?

Reply:
PS this is the code from super where the random should be fine. => Or??

```
    public @Nullable Thing createThing(ThingTypeUID thingTypeUID, Configuration configuration,
            @Nullable ThingUID thingUID, @Nullable ThingUID bridgeUID) {
        ThingUID effectiveUID = thingUID != null ? thingUID : ThingFactory.generateRandomThingUID(thingTypeUID);
        ThingType thingType = getThingTypeByUID(thingTypeUID);
        if (thingType != null) {
            return ThingFactory.createThing(thingType, effectiveUID, configuration, bridgeUID,
                    getConfigDescriptionRegistry());
        } else {
            return null;
        }
    }
```

Reply:
I did similar to what you had in the original binding. Not sure why it is needed. So @lolodomo please check